### PR TITLE
Ruby-Catalog

### PR DESCRIPTION
**Pull request by AmineDerbal**
In this pull request ♻️ 

video presentation link : https://drive.google.com/file/d/1xNiR2_3IRNBrY_58pyiZvZzPmbWpeSl-/view?usp=sharing

**Group** 

- Setup project 👍 
- create Item class 👍 
- create main.rb 👍 

@Leeoasis 

- create book class 👍 
- create label class 👍 
- list all books 👍 
- list all labels 👍 
- add book 👍 
- preserve data for books and labels 👍 
- implement tests for book and label class 👍 
- create database schema for book and label 👍 

@AmineDerbal 
- create music album class 👍 
- create genre class 👍 
- list all music albums 👍 
- list all genres 👍 
- add music album 👍 
- preserve data for music albums and genres 👍 
- implement tests for music album and genre class 👍 
- create database schema for music album and genre 👍 

@Lornakaboro 
- create game class 👍 
- create author class 👍 
- list all games 👍 
- list all author 👍 
- add game 👍 
- preserve data for games and authors 👍 
- implement tests for game and author class 👍 
- create database schema for game and author 👍

### CODE REVIEW

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @Team👋👋👋, 

tryagain


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Gitflow have been used👍.
- [x] Project has a descriptive and meaningful PR title and summary👍.
- [x] Project is professionally documented👍.
- [x] Unit tests have been written for each methods👍.
- [x] Each class from the UML diagram have been implemented in seperate files👍.
- [x] All aggregation of parent-child relationships have been written👍.
- [x] The `main.rb` file have been used as the entry point👍.


## Required Changes ♻️

- [ ] Nice job so far guys👍. Your project looks great😉. For your video presentation, we are required to record a video whose length is not more than 5mins😉. currently, your video presentation is about 8 minutes 30 seconds🥲. Please let's try to stick to 5mins as required👍.

image


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/item.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  private
+
+  def can_be_archived?
+    @publish_date = Date.parse(@publish_date) unless @publish_date.is_a?(Date)
+    today = Date.today
+    ten_years_ago = (today - (10 * 365))
+    @publish_date < ten_years_ago
+  end
+end
```
[the remark this answers is not in the document]
Comment (AmineDerbal) [reply classes/item.rb]: Understood It will be done 👍

**Comment by AmineDerbal** [comment]
- implement the one to many relationship between item and {genre, author, label) in the item class as suggested by the code reviewer 👍 
- make a new video presentation of less than 5 minutes link : https://drive.google.com/file/d/16VbNGHqAvV6UIGr_1glezgBkX-qeAZCP/view?usp=sharing 👍

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👏 Great job recording another video and highlighting a piece of code you are proud of.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**